discovery/seata-discovery-sofa/src/main/java/io/seata/discovery/registry/sofa/SofaRegistryServiceImpl.java

```diff
@@ -0,0 +1,323 @@
+                    String address = registryProps.getProperty(PRO_SERVER_ADDR_KEY);
+                    final String portStr = StringUtils.substringAfter(address, HOST_SEPERATOR);
+
+                    RegistryClientConfig config = DefaultRegistryClientConfigBuilder.start()
+                        .setAppName(getApplicationName())
+                        .setDataCenter(registryProps.getProperty(PRO_DATACENTER_KEY))
+                        .setZone(registryProps.getProperty(PRO_REGION_KEY))
+                        .setRegistryEndpoint(StringUtils.substringBefore(address, HOST_SEPERATOR))
+                        .setRegistryEndpointPort(Integer.parseInt(portStr)).build();
+
+                    registryClient = new DefaultRegistryClient(config);
+                    ((DefaultRegistryClient)registryClient).init();
+                }
+            }
+        }
+        return registryClient;
+    }
+
+    @Override
+    public void subscribe(String cluster, SubscriberDataObserver listener) throws Exception {
+        SubscriberRegistration subscriberRegistration = new SubscriberRegistration(cluster, listener);
+        subscriberRegistration.setScopeEnum(ScopeEnum.global);
+        subscriberRegistration.setGroup(registryProps.getProperty(PRO_GROUP_KEY));
+
+        LISTENER_SERVICE_MAP.putIfAbsent(cluster, new ArrayList<>());
+        LISTENER_SERVICE_MAP.get(cluster).add(listener);
+        getRegistryInstance().register(subscriberRegistration);
+    }
+
+    @Override
+    public void unsubscribe(String cluster, SubscriberDataObserver listener) throws Exception {
+        getRegistryInstance().unregister(cluster, registryProps.getProperty(PRO_GROUP_KEY), RegistryType.SUBSCRIBER);
+    }
+
+    @Override
+    public List<InetSocketAddress> lookup(String key) throws Exception {
+        Configuration config = ConfigurationFactory.getInstance();
+        String clusterName = config.getConfig(PREFIX_SERVICE_ROOT + CONFIG_SPLIT_CHAR + PREFIX_SERVICE_MAPPING + key);
+        if (null == clusterName) {
+            return null;
+        }
+        if (!LISTENER_SERVICE_MAP.containsKey(clusterName)) {
+            subscribe(clusterName, new SubscriberDataObserver() {
+                @Override
+                public void handleData(String dataId, UserData data) {
+                    Map<String, List<String>> instances = data.getZoneData();
+                    if (null == instances && null != CLUSTER_ADDRESS_MAP.get(clusterName)) {
+                        CLUSTER_ADDRESS_MAP.remove(clusterName);
+                    } else {
+                        List<InetSocketAddress> tranformData = flatData(instances);
+                        List<InetSocketAddress> newAddressList = new ArrayList<>();
+                        newAddressList.addAll(tranformData);
+                        CLUSTER_ADDRESS_MAP.put(clusterName, newAddressList);
+                    }
+                    respondRegistries.countDown();
```

Comment:
I think respondRegistries  should be local variable，how to support multi-cluster?

Reply:
yes，changed